bundles/org.openhab.binding.ism8/src/main/java/org/openhab/binding/ism8/server/Server.java

```diff
@@ -0,0 +1,281 @@
+            try {
+                this.handleCommunication();
+                if (this.startRetries > 10) {
+                    Thread.sleep(6000);
+                }
+            } catch (InterruptedException e) {
+                logger.debug("Thread interrupted");
+            } catch (Exception e) {
+                logger.warn("Error Handle Communication - restart communication. {}", e.getMessage(), e);
+                this.startRetries++;
+            }
+        }
+    }
+
+    /**
+     * Stops the server
+     *
+     */
+    public void stopServerThread() {
+        this.interrupt();
+        this.stopServer();
+    }
+
+    /**
+     * Adds a data-point change listener to the server
+     *
+     */
+    public void addDataPointChangeListener(IDataPointChangeListener listener) {
+        if (this.changeListener == null) {
+            this.changeListener = listener;
+        }
+    }
+
+    /**
+     * Adds a data-point to the server
+     *
+     */
+    public void addDataPoint(int id, String knxType, String description) {
+        IDataPoint dp = DataPointFactory.createDataPoint(id, knxType, description);
+        if (dp != null) {
+            for (IDataPoint dataPoint : this.dataPoints) {
+                if (dataPoint.getId() == dp.getId()) {
+                    return;
+                }
+            }
+            this.dataPoints.add(dp);
+        }
+    }
+
+    /**
+     * Sends the data to the ISM8 partner
+     *
+     */
+    public void sendData(byte[] data) throws IOException {
+        Socket clientSocket = this.client;
+        if (clientSocket != null && clientSocket.isConnected() && data.length > 0) {
+            OutputStream stream = clientSocket.getOutputStream();
+            stream.write(data);
+            logger.debug("Data sent: {}", this.printBytes(data));
+        }
+    }
+
+    private void stopServer() {
+        logger.info("Stop Ism8 server.");
+        try {
+            ServerSocket serverSock = this.serverSocket;
+            if (serverSock != null) {
+                serverSock.close();
+            }
+
+            Socket clientSocket = this.client;
+            if (clientSocket != null) {
+                if (clientSocket.isConnected()) {
+                    clientSocket.getInputStream().close();
+                }
+
+                clientSocket.close();
+                this.client = null;
+            }
+        } catch (Exception e) {
+            logger.warn("Error stopping Communication. {}", e.getMessage(), e);
+        }
+    }
+
+    private void handleCommunication() {
+        try {
+            logger.info("Waiting for connection in port {}.", this.getPort());
+            IDataPointChangeListener listener = this.changeListener;
+            if (listener != null) {
+                listener.connectionStatusChanged(ThingStatus.OFFLINE);
+            }
+            
+            ServerSocket serverSock =new java.net.ServerSocket(this.getPort());
```

Comment:
Can you use the formatter to format the code? The package name could be removed.